### ISSUE TITLE
what will happen setAutoPlayAnimations(false) for gif images 

### ISSUE THREAD

**Issue by parane** [facebook/fresco#1326]
when setAutoPlayAnimations(false) (**not supported for caching** )  , it will only take the first frame as preview image. But I think it will be decode whole frame (may be need additional resources and not supported for caching also). and just show the first frame  so setAutoPlayAnimations(false) can we show static image without decode whole frame   ? 

may be this some already discuss topic : 
https://github.com/facebook/fresco/pull/1304

**Comment by huangjunkun** [comment]
You should do the following, only take the first frame as preview image. 
  ```
/**
   * Force image to be rendered as a static image, even if it is an animated format.
   *
   * This flag will force animated GIFs to be rendered as static images
   */
  public final boolean forceStaticImage;
```

> com.facebook.imagepipeline.request.ImageRequestBuilder#setImageDecodeOptions
> com.facebook.imagepipeline.common.ImageDecodeOptionsBuilder#setForceStaticImage

**Comment by parane** [comment]
i have tried but same memory consumption for decode all images
```
ImageDecodeOptionsBuilder b = new ImageDecodeOptionsBuilder();
        b.setDecodeAllFrames(false);
        ImageDecodeOptions imageDecodeOptions=new ImageDecodeOptions(b);
        ImageRequest request =  ImageRequestBuilder.newBuilderWithSource(animatedGifUri).setImageDecodeOptions(imageDecodeOptions).setLocalThumbnailPreviewsEnabled(true).build();

```

**Comment by huangjunkun** [comment]
@parane  
`b.setDecodeAllFrames(false);`  that is decode all frames before playing animated image.
Do you want to decode the preview frame (a static image), not all frames (animated image) ?

Yes, and you should **set true to ImageDecodeOptionsBuilder#setForceStaticImage**.

btw. Please update the Fresco v0.11.0.

**Comment by oprisnik** [comment]
We've released [Fresco v1.4.0](https://github.com/facebook/fresco/releases/tag/v1.4.0), which includes a new implementation for animated GIFs that is way more performant. Let us know if you're still experiencing problems with 1.4.0.
